File: 1049.py
#!/usr/bin/env python

# runtime of python version is more than 10 ms 
# because of the implementation 
if __name__ == '__main__':
    string = raw_input()
    n = int(string)
    r = 1
    counts_for_1 = 0
    ret, rem = divmod(n, r)
    while ret != 0:
        left = ret / 10
        current = ret % 10
        right = rem
        if current == 1:
            counts_for_1 += right + 1 + left * r
        elif current > 1:
            counts_for_1 += (left + 1) * r
        else:
            counts_for_1 += left * r
        r *= 10
        ret, rem = divmod(n, r) 
    print counts_for_1
    print r

File: 1049.cpp
#include<stdio.h>

int main()
{
    int n;
    long long sum = 0;
    scanf("%d", &n);
    int left, current, right;
    int r = 1;
    while (n/r) {
        left = n/(r*10);
        current = (n/r)%10;
        right = n - (n/r)*r;
        if (current == 1) {
            sum += right + 1 + left * r;
        }else if (current > 1 ) {
            sum += (left + 1)*r;
        }else {
            sum += left * r;
        }

        r *= 10;
    }

    printf("%ld\n", sum);
}
